source/patterns/@aws-solutions-constructs/aws-openapigateway-lambda/lib/openapi-helper.ts

```diff
@@ -0,0 +1,185 @@
+ */
+
+/*
+ *  The functions found here in the core library are for internal use and can be changed
+ *  or removed outside of a major release. We recommend against calling them directly from client code.
+ */
+
+/*
+ * This file is core openapi functionality and should ideally be in the core library. Since
+ * that causes a circular reference with the resources library we have left it here for now
+ * in the interest of getting these updates out faster
+ */
+
+import * as apigateway from 'aws-cdk-lib/aws-apigateway';
+import * as lambda from 'aws-cdk-lib/aws-lambda';
+import * as s3 from 'aws-cdk-lib/aws-s3';
+import { Asset } from 'aws-cdk-lib/aws-s3-assets';
+import { Aws, Duration } from "aws-cdk-lib";
+import { Construct } from 'constructs';
+import * as resources from '@aws-solutions-constructs/resources';
+
+/**
+ * The ApiIntegration interface is used to correlate a user-specified id with either a existing lambda function or set of lambda props.
+ *
+ * See the 'Overview of how the OpenAPI file transformation works' section of the README.md for more details on its usage.
+ */
+export interface ApiIntegration {
+  /**
+   * Id of the ApiIntegration, used to correlate this lambda function to the api integration in the open api definition.
+   *
+   * Note this is not a CDK Construct ID, and is instead a client defined string used to map the resolved lambda resource with the OpenAPI definition.
+   */
+  readonly id: string;
+  /**
+   * The Lambda function to associate with the API method in the OpenAPI file matched by id.
+   *
+   * One and only one of existingLambdaObj or lambdaFunctionProps must be specified, any other combination will cause an error.
+   */
+  readonly existingLambdaObj?: lambda.Function;
+  /**
+   * Properties for the Lambda function to create and associate with the API method in the OpenAPI file matched by id.
+   *
+   * One and only one of existingLambdaObj or lambdaFunctionProps must be specified, any other combination will cause an error.
+   */
+  readonly lambdaFunctionProps?: lambda.FunctionProps;
+}
+
+/**
+ * Helper object to map an ApiIntegration id to its resolved lambda.Function. This type is exposed as a property on the instantiated construct.
+ */
+export interface TokenToFunctionMapping {
+  /**
+   * Id of the ApiIntegration, used to correlate this lambda function to the api integration in the open api definition.
+   */
+  readonly id: string;
+  /**
+   * The instantiated lambda.Function.
+   */
+  readonly lambdaFunction: lambda.Function;
+}
+
+export interface OpenApiProps {
+  readonly apiDefinitionAsset?: Asset,
+  readonly apiJsonDefinition?: any,
+  readonly apiDefinitionBucket?: s3.IBucket,
+  readonly apiDefinitionKey?: string,
+  readonly apiIntegrations: ApiIntegration[]
+}
+
+/**
+ * @internal This is an internal core function and should not be called directly by Solutions Constructs clients.
+ */
+export function CheckOpenapiProps(props: OpenApiProps) {
```

Comment:
Nit - `CheckOpenApiProps` to match the casing of `OpenApi` elsewhere in this code.